bin/pt-stalk

```diff
@@ -796,9 +801,65 @@ collect() {
    local d="$1"  # directory to save results in
    local p="$2"  # prefix for each result file
 
-   local mysqld_pid=""
+   local cnt=$(($OPT_RUN_TIME / $OPT_SLEEP_COLLECT))
+   
+   if [ ! "$OPT_SYSTEM_ONLY" ]; then
+      local mysqld_pid=""
+      local mysql_version=""
+      local mysql_error_log=""
+      local tail_error_log_pid=""
+      local have_lock_waits_table=""
+      local have_oprofile=""
+      local mysqladmin_pid=""
+      local mutex=""
+      local tcpdump_pid=""
+      local ps_instrumentation_enabled=""
+
+      collect_mysql_data_one
+   fi
+
+   if [ ! "$OPT_MYSQL_ONLY" ]; then 
+      collect_system_data
+   fi 
+
+   log "Loop start: $(date +'TS %s.%N %F %T')"
+   local start_time=$(date +'%s')
+   local curr_time=$start_time
+   local ts="$(date +"TS %s.%N %F %T")"
+
+   while [ $((curr_time - start_time)) -lt $OPT_RUN_TIME ]; do
+      if [ ! "$OPT_MYSQL_ONLY" ]; then
+         collect_system_data_loop
+      fi
+
+      if [ ! "$OPT_SYSTEM_ONLY" ]; then
+         collect_mysql_data_loop
+      fi
+
+      curr_time=$(date +'%s')
+   done
+   log "Loop end: $(date +'TS %s.%N %F %T')"
+
+   if [ ! "$OPT_SYSTEM_ONLY" ]; then
+      collect_mysql_data_two
+   fi
+
+   hostname > "$d/$p-hostname"
+
+   wait_for_subshells $OPT_RUN_TIME
+   kill_all_subshells
+   for file in "$d/$p-"*; do
+      if [ -z "$(grep -v '^TS ' --max-count 10 "$file")" ]; then
+         log "Removing empty file $file";
+         rm "$file"
+      fi
+   done
+}
+
+collect_mysql_data_one() {
    if [ ! "$OPT_MYSQL_ONLY" ]; then
-      mysqld_pid=$(_pidof mysqld | awk '{print $1; exit;}')
+      port=$($CMD_MYSQL $EXT_ARGV -ss -e 'SELECT @@port')
```

Comment:
ok, is this some kind of improvement? Should we separate this to another ticket, so it could be more descriptive. Because adding it in the scope of this fix - it could be hard to figure out later why it was done.

Reply:
This is the fix for https://jira.percona.com/browse/PT-1398